src/query/functions/src/scalars/map.rs

```diff
@@ -244,4 +258,134 @@ pub fn register(registry: &mut FunctionRegistry) {
+                            arg_type,
+                            DataType::String
+                                | DataType::Number(_)
+                                | DataType::Decimal(_)
+                                | DataType::Date
+                                | DataType::Timestamp
+                        )
+                    },
+                    |key_type| arg_type == key_type,
+                )
+            }),
+        };
+        if !key_match {
+            return None;
+        }
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_pick".to_string(),
+                args_type: args_type.to_vec(),
+                return_type: args_type[0].clone(),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(move |_, _| FunctionDomain::Full),
+                eval: Box::new(map_pick_fn_vec),
+            },
+        }))
+    });
+
+    fn map_pick_fn_vec(args: &[ValueRef<AnyType>], _: &mut EvalContext) -> Value<AnyType> {
+        let len = args.iter().find_map(|arg| match arg {
+            ValueRef::Column(col) => Some(col.len()),
+            _ => None,
+        });
+
+        let source_data_type = match args.first().unwrap() {
+            ValueRef::Scalar(s) => s.infer_data_type(),
+            ValueRef::Column(c) => c.data_type(),
+        };
+
+        let source_map = match &args[0] {
+            ValueRef::Scalar(s) => match s {
+                ScalarRef::Map(cols) => {
+                    KvPair::<GenericType<0>, GenericType<1>>::try_downcast_column(cols).unwrap()
+                }
+                ScalarRef::EmptyMap => {
+                    KvPair::<GenericType<0>, GenericType<1>>::try_downcast_column(
+                        &Column::EmptyMap { len: 0 },
+                    )
+                    .unwrap()
+                }
+                _ => unreachable!(),
+            },
+            ValueRef::Column(Column::Map(c)) => {
+                KvPair::<GenericType<0>, GenericType<1>>::try_downcast_column(&c.values).unwrap()
+            }
+            _ => unreachable!(),
+        };
+
+        let mut builder: ArrayColumnBuilder<KvPair<GenericType<0>, GenericType<1>>> =
+            ArrayType::create_builder(
+                args.len() - 1,
+                source_data_type.as_map().unwrap().as_tuple().unwrap(),
+            );
+        let select_keys = match &args[1] {
+            ValueRef::Scalar(ScalarRef::Array(arr)) if args.len() == 2 => {
+                arr.iter().collect::<Vec<_>>()
+            }
+            _ => args[1..]
+                .iter()
+                .map(|arg| arg.as_scalar().unwrap().clone())
+                .collect::<Vec<_>>(),
+        };
+        for key_arg in select_keys {
+            if let Some((k, v)) = source_map.iter().find(|(k, _)| k == &key_arg) {
+                builder.put_item((k.clone(), v.clone()));
+            }
+        }
+        builder.commit_row();
```

Comment:
those code logic seems not correct, `arg.as_scalar()` will panic if the value is a column. and the builder need to work for every row, not only the first value.